code/game/objects/structures/training_machine.dm

```diff
@@ -0,0 +1,234 @@
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "TrainingMachine", name)
+		ui.open()
+
+/obj/structure/training_machine/ui_data(mob/user)
+	var/list/data = list()
+	data["range"] = range
+	data["movespeed"] = move_speed
+	data["moving"] = moving
+	return data
+
+/obj/structure/training_machine/ui_act(action, params)
+	if(..())
+		return
+	switch(action)
+		if("toggle")
+			toggle()
+		if("range")
+			var/range_input = params["range"]
+			range = clamp(range_input, MIN_RANGE, MAX_RANGE)
+		if("movespeed")
+			var/range_input = params["movespeed"]
+			move_speed = clamp(range_input, MIN_SPEED, MAX_SPEED)
+
+/obj/structure/training_machine/attack_hand(mob/user)
+	ui_interact(user)
+
+/obj/structure/training_machine/attackby(var/obj/item/target/target, mob/user)
+	. = ..()
+	if (!istype(target))
+		return
+	if (length(buckled_mobs))
+		return
+	if (pinned_target) //Swap out the old with the new
+		vis_contents.Cut()
+		user.put_in_hands(pinned_target)
+	pinned_target = attacking_obj
+	pinned_target.forceMove(src)
+	vis_contents += pinned_target
+	handle_density()
+	to_chat(user, "<span class='notice'>You attach the target to the seat.</span>")
+	playsound(src, "rustle", 50, TRUE)
+
+/obj/structure/training_machine/AltClick(mob/user)
+	. = ..()
+	if (pinned_target)
+		vis_contents.Cut()
+		user.put_in_hands(pinned_target)
+		pinned_target = null
+		handle_density()
+		to_chat(user, "<span class='notice'>You remove the target from the seat.</span>")
+		playsound(src, "rustle", 50, TRUE)
+
+/obj/structure/training_machine/proc/toggle()
+	if (moving)
+		moving = FALSE
+	else
+		start_moving()
+
+/obj/structure/training_machine/proc/stop_moving(message = "Ending training simulation.")
+	moving = FALSE
+	starting_turf = null
+	say(message)
+	playsound(src,'sound/machines/synth_no.ogg',50,FALSE)
+
+/obj/structure/training_machine/proc/start_moving()
+	moving = TRUE
+	starting_turf = get_turf(src)
+	say("Beginning training simulation.")
+	playsound(src,'sound/machines/triple_beep.ogg',50,FALSE)
+	do_movement()
+
+/obj/structure/training_machine/proc/do_movement()
+	var/turf/current_turf = get_turf(src)
+	if (!moving || !starting_turf || isspaceturf(current_turf))
+		stop_moving()
+		return
+	if (current_turf == target_position) //We've reached our target turf, now find a new one
+		target_position = null
+	if (!target_position)
+		target_position = find_target_position()
+		if (!target_position)
+			stop_moving("ERROR! Cannot calculate suitable movement path.")
+	var/turf/nextStep = get_step_towards(src, target_position)
+	if (!Move(nextStep))
+		target_position = null //We couldn't move towards the target turf, so find a new target turf
+	if (obj_flags & EMAGGED && world.time > last_attack_time + attack_cooldown)
+		var/mob/living/carbon/target = locate() in view(1, src)
+		if (target?.stat == CONSCIOUS)
+			do_attack_animation(target, null, secret_evil_toolbox)
+			target.adjustBruteLoss(secret_evil_toolbox.force)
```

Comment:
this ignores armor...

Reply:
push it to the limit

Reply:
9 hits to a drop everyone knows the rules